go/cmd/dolt/commands/init.go

```diff
@@ -154,7 +155,15 @@ func (cmd InitCmd) Exec(ctx context.Context, commandStr string, args []string, d
 	}
 
 	requiresFunHash := apr.Contains(funHashFlag)
-	err := dEnv.InitRepoWithTime(context.Background(), types.Format_Default, name, email, initBranch, t, requiresFunHash)
+	commitMeta := func() datas.CommitMetaGenerator {
+		if requiresFunHash {
+			return datas.MakeFunCommitMetaGenerator(name, email, t)
+		} else {
+			return datas.MakeCommitMetaGenerator(name, email, t)
+		}
+	}()
+
+	err := dEnv.InitRepoWithCommitMeta(context.Background(), types.Format_Default, initBranch, commitMeta)
```

Comment:
Ohhh... I like this new approach a lot! 

Just one suggestion... how about renaming the function to: `InitRepoWithCommitMetaGenerator`. I think that's a better, more self-descriptive name because there is another type called `CommitMeta`, so I initially assumed this function was taking in a `CommitMeta` instance (plus the var was named `commitMeta`). Using `CommitMetaGenerator` would make it a little clearer and easier to understand quickly.